src/middleware.ts

```diff
@@ -9,7 +9,11 @@ import { NextRequest, NextResponse } from "next/server";
  * However, their middleware doesn't yet work with database sessions.
  */
 export function middleware(request: NextRequest) {
-  if (!request.cookies.get("next-auth.session-token")) {
+  const sessionToken =
+    request.cookies.get("next-auth.session-token") ??
+    request.cookies.get("__Secure-next-auth.session-token");
```

Comment:
All the fascinating intricacies of various frameworks.
Thanks for fixing this, @codetheweb